Add OpenMetrics documentation. (#3878)
temporalio/documentation: docs/production-deployment/cloud/metrics/openmetrics/index.mdx

---
id: index
title: Temporal Cloud OpenMetrics
sidebar_label: OpenMetrics
description: Get detailed insights into your Temporal Cloud Namespace metrics using your own observability tool. Access data with a CA certificate and retain raw metrics for seven days.
slug: /cloud/metrics/openmetrics/
toc_max_heading_level: 4
keywords:
  - explanation
  - how-to
  - introduction
  - metrics
  - observability
  - promentheus
  - reference
  - tcld cli
  - temporal cloud
tags:
  - Metrics
  - Observability
  - Temporal Cloud
---

:::tip SUPPORT, STABILITY, and DEPENDENCY INFO

> Temporal Cloud OpenMetrics support is available in  [Public Preview](/evaluate/development-production-features/release-stages#public-preview).

Reviewer: How about:
Use cases that require scraping that exceeds limits are likely to be priced in the future.

slug: /cloud/metrics/openmetrics/
toc_max_heading_level: 4
keywords:
  - explanation
  - how-to
  - introduction
  - metrics
  - observability
  - promentheus
  - reference
  - tcld cli
  - temporal cloud
tags:
  - Metrics
  - Observability
  - Temporal Cloud
---

:::tip SUPPORT, STABILITY, and DEPENDENCY INFO

Temporal Cloud OpenMetrics support is available in  [Public Preview](/evaluate/development-production-features/release-stages#public-preview).

:::

:::tip PRICING

Future pricing may apply to high-volume usage that exceeds standard [limits](/production-deployment/cloud/metrics/openmetrics/api-reference#api-limits).

:::

Temporal Cloud's [OpenMetrics](https://openmetrics.io/) Endpoint provides operational metrics for your Temporal Cloud workloads in industry-standard Prometheus format, enabling comprehensive monitoring across Namespaces, Workflows, and Task Queues with your existing observability stack.

## Quick Links
* [API Documentation](/production-deployment/cloud/metrics/openmetrics/api-reference) - Endpoint specification and integration examples

> * [Metrics Reference](/production-deployment/cloud/metrics/openmetrics/metrics-reference) - Complete catalog with descriptions and labels

Reviewer 2: ```suggestion
* [Metrics Reference](/production-deployment/cloud/metrics/openmetrics/metrics-reference) - Complete catalog of all metrics with descriptions and labels
```

> * [Migration Guide](/production-deployment/cloud/metrics/openmetrics/migration-guide) - Transition from the query endpoint

Reviewer 2: ```suggestion
* [Migration Guide](/production-deployment/cloud/metrics/openmetrics/migration-guide) - Guide on how to transition from the query endpoint
```

> * View [integration examples](https://github.com/temporal-community/cloud-metrics-scrape-examples) for various platforms

Reviewer 2: ```suggestion
* [Integration examples](https://github.com/temporal-community/cloud-metrics-scrape-examples) - Practical examples for various platforms
```

## Overview
Temporal Cloud OpenMetrics exposes 30+ metrics covering workflow lifecycles, task queue operations, service performance, and system limits. **All metrics are aggregated over one-minute windows and available for scraping within two minutes**.

* [Set up authentication and scraping](/production-deployment/cloud/metrics/openmetrics/api-reference) with the API documentation. 
* Browse the [complete metrics catalog](/production-deployment/cloud/metrics/openmetrics/metrics-reference) for descriptions and labels. 
* Teams using the query endpoint should review the [migration guide](/production-deployment/cloud/metrics/openmetrics/migration-guide).

## API key authentication
Create a service account with "Metrics Read-Only" role, generate an API key, and start scraping immediately - no certificate rotation or distribution required.

## Global endpoint
Single endpoint at `metrics.temporal.io` which serves all metrics across your entire account with API key authentication and standard HTTPS.

## Namespace and metric filtering

> Query parameters enable selective scraping to manage data volume and costs, supporting wildcards for flexible namespace selection and specific metric filtering.

Reviewer 2: ```suggestion
You can use query parameters to enable selective scraping to manage data volume and costs, which support wildcards for flexible namespace selection and specific metric filtering.
```
